cmd/cluster-controller/main.go

```diff
@@ -0,0 +1,150 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-capi-operator/pkg/controllers/cluster"
+	"github.com/openshift/cluster-capi-operator/pkg/operatorstatus"
+	"github.com/openshift/cluster-capi-operator/pkg/util"
+)
+
+var (
+	scheme               = runtime.NewScheme()
+	leaderElectionConfig = config.LeaderElectionConfiguration{
+		LeaderElect:       true,
+		LeaseDuration:     util.LeaseDuration,
+		RenewDeadline:     util.RenewDeadline,
+		RetryPeriod:       util.RetryPeriod,
+		ResourceName:      "cluster-controller-leader",
+		ResourceNamespace: "openshift-cluster-api",
+	}
+)
+
+const (
+	defaultProvidersLocation      = "./providers-list.yaml"
+	releaseVersionEnvVariableName = "RELEASE_VERSION"
+	unknownVersionValue           = "unknown"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(configv1.AddToScheme(scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(scheme))
+	utilruntime.Must(operatorv1.AddToScheme(scheme))
+	utilruntime.Must(awsv1.AddToScheme(scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme))
+
+	// +kubebuilder:scaffold:scheme
+}
+
+func main() {
+	klog.InitFlags(nil)
+
+	ctrl.SetLogger(klogr.New())
+
+	metricsAddr := flag.String(
+		"metrics-bind-address",
+		":8080",
+		"Address for hosting metrics",
+	)
+	healthAddr := flag.String(
+		"health-addr",
+		":9440",
+		"The address for health checking.",
+	)
+
+	providerFile := flag.String(
+		"providers-yaml",
+		defaultProvidersLocation,
+		"The location of supported providers for CAPI",
+	)
+
+	// Once all the flags are regitered, switch to pflag
+	// to allow leader lection flags to be bound
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	options.BindLeaderElectionFlags(&leaderElectionConfig, pflag.CommandLine)
+	pflag.Parse()
+
+	ctrl.SetLogger(klogr.New().WithName("CAPIClusterController"))
+
+	syncPeriod := 10 * time.Minute
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                  scheme,
+		SyncPeriod:              &syncPeriod,
+		MetricsBindAddress:      *metricsAddr,
+		Port:                    9443,
+		HealthProbeBindAddress:  *healthAddr,
+		LeaderElectionNamespace: leaderElectionConfig.ResourceNamespace,
+		LeaderElection:          leaderElectionConfig.LeaderElect,
+		LeaseDuration:           &leaderElectionConfig.LeaseDuration.Duration,
+		LeaderElectionID:        leaderElectionConfig.ResourceName,
+		RetryPeriod:             &leaderElectionConfig.RetryPeriod.Duration,
+		RenewDeadline:           &leaderElectionConfig.RenewDeadline.Duration,
+	})
+	if err != nil {
+		klog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	supportedProviders, err := util.ReadProvidersFile(*providerFile)
+	if err != nil {
+		klog.Error(err, "unable to get providers from file", "name", *providerFile)
+		os.Exit(1)
+	}
+
+	if err = (&cluster.ClusterReconciler{
```

Comment:
Don't shadow the previous err declaration here
```suggestion
	if err := (&cluster.ClusterReconciler{
```